src/Utilities/TagsModal.js

```diff
@@ -143,7 +143,11 @@ const TagsModal = ({ filterTagsBy, onToggleModal, onApply, getTags }) => {
       bulkSelect={{ id: 'bulk-select-tags' }}
       title={
         activeSystemTag
-          ? `${activeSystemTag.display_name} (${tagsCount})`
+          ? `${
+              activeSystemTag.display_name ||
```

Comment:
Yeah, I believe this is the problematic code:
```
  const activeSystemTag = useSelector(
    ({ entities, entityDetails }) =>
      entities?.activeSystemTag || entityDetails?.entity
```
the problem is that `entities?.activeSystemTag` always defined and has a "truthy" value, thus `entityDetails?.entity` is not used (but it must be picked instead when on inventory details page)

Reply:
Alright, should work now, replaced it with

```
  const activeSystemTag = useSelector(({ entities, entityDetails }) =>
    entities?.activeSystemTag?.created
      ? entities.activeSystemTag
      : entityDetails?.entity
  );
```

instead of `created` it could be any other property that is only there if it's describing system. 

Have to say that I don't understand why sometimes it's in `entities` and sometimes in `entityDetails` (list of systems vs detail of system?), but it probably has a reason? Would have to inspect the codebase more :sweat_smile: .

It seems to work now, I tried it locally :shrug: